apps/frontend/app/root.tsx

```diff
@@ -105,6 +106,19 @@ export const loader = unstable_defineLoader(async ({ request }) => {
 });
 
 const DefaultHeadTags = () => {
+	const [isIOS18, setIsIOS18] = useState(false);
+
+	useEffect(() => {
```

Comment:
Instead of useEffect+useState combo, it would be better to use just useMemo here.

Reply:
Though probably not useful here, mantine has a similar hook: https://mantine.dev/hooks/use-os. Could that be used somehow?

Reply:
useMemo wouldnt work as window isnt avaliable during SSR AFAIK

Reply:
It works fine. Remix takes care of it. Eg: https://github.com/IgnisDa/ryot/blob/72c800895cebd8913c44e8ce33fbbe8c6802b3cd/apps/frontend/app/routes/_dashboard._index.tsx#L628

Reply:
On futher thought, i think the entire thing can be lifted out of state and to the `loader` in `root.tsx`. THe headers can be used to determine the UA.

Reply:
```
night@Pandemos:~$ curl -s -A "Mozilla/5.0 (iPhone; CPU iPhone OS 18_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1" http://10.0.0.68:8000/auth?redirectTo=%2F | grep -o '<link rel="apple-touch-icon"[^>]*>'
<link rel="apple-touch-icon" href="/icon-192x192.png"/>
night@Pandemos:~$ curl -s -A "Mozilla/5.0 (iPhone; CPU iPhone OS 17_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Mobile/15E148 Safari/604.1" http://10.0.0.68:8000/auth?redirectTo=%2F | grep -o '<link rel="apple-touch-icon"[^>]*>'
<link rel="apple-touch-icon" href="/apple-touch-icon.png"/>
night@Pandemos:~$ curl -s -A "Mozilla/5.0 (Linux; Android 10; SM-G960U) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Mobile Safari/537.36" http://10.0.0.68:8000/auth?redirectTo=%2F | grep -o '<link rel="apple-touch-icon"[^>]*>'
<link rel="apple-touch-icon" href="/apple-touch-icon.png"/>
night@Pandemos:~$ curl -s -A "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0" http://10.0.0.68:8000/auth?redirectTo=%2F | grep -o '<link rel="apple-touch-icon"[^>]*>'
<link rel="apple-touch-icon" href="/apple-touch-icon.png"/>
night@Pandemos:~$ curl -s -A "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36" http://10.0.0.68:8000/auth?redirectTo=%2F | grep -o '<link rel="apple-touch-icon"[^>]*>'
<link rel="apple-touch-icon" href="/apple-touch-icon.png"/>
```